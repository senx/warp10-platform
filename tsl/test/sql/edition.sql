-- This file and its contents are licensed under the Timescale License.
-- Please see the included NOTICE for copyright information and
-- LICENSE-TIMESCALE for a copy of the license.

SELECT _timescaledb_internal.tsl_loaded();
SELECT _timescaledb_internal.enterprise_enabled();
select * from timescaledb_information.license;

-- changing licenses requires superuser privleges
\set ON_ERROR_STOP 0
SET timescaledb.license_key='CommunityLicense';
SELECT _timescaledb_internal.current_license_key();
SELECT _timescaledb_internal.tsl_loaded();
SELECT _timescaledb_internal.enterprise_enabled();
\set ON_ERROR_STOP 1

\c :TEST_DBNAME :ROLE_SUPERUSER
SELECT _timescaledb_internal.current_license_key();
SELECT _timescaledb_internal.tsl_loaded();
SELECT _timescaledb_internal.enterprise_enabled();

\set ON_ERROR_STOP 0
SET timescaledb.license_key='ApacheOnly';
SELECT _timescaledb_internal.current_license_key();
SELECT _timescaledb_internal.tsl_loaded();
SELECT _timescaledb_internal.enterprise_enabled();
\set ON_ERROR_STOP 1


SET timescaledb.license_key='CommunityLicense';
SELECT _timescaledb_internal.current_license_key();
SELECT _timescaledb_internal.tsl_loaded();
SELECT _timescaledb_internal.enterprise_enabled();
select * from timescaledb_information.license;

SET timescaledb.license_key=Default;
SELECT _timescaledb_internal.current_license_key();
SELECT _timescaledb_internal.tsl_loaded();
SELECT _timescaledb_internal.enterprise_enabled();
select * from timescaledb_information.license;
