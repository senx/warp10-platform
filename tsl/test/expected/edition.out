-- This file and its contents are licensed under the Timescale License.
-- Please see the included NOTICE for copyright information and
-- LICENSE-TIMESCALE for a copy of the license.
SELECT _timescaledb_internal.tsl_loaded();
 tsl_loaded 
------------
 t
(1 row)

SELECT _timescaledb_internal.enterprise_enabled();
 enterprise_enabled 
--------------------
 t
(1 row)

select * from timescaledb_information.license;
  edition   | expired |       expiration_time        
------------+---------+------------------------------
 enterprise | t       | Sun Sep 30 17:00:00 2018 PDT
(1 row)

-- changing licenses requires superuser privleges
\set ON_ERROR_STOP 0
SET timescaledb.license_key='CommunityLicense';
ERROR:  permission denied to set parameter "timescaledb.license_key"
SELECT _timescaledb_internal.current_license_key();
ERROR:  must be superuser or a member of pg_read_all_settings to examine the license key
SELECT _timescaledb_internal.tsl_loaded();
 tsl_loaded 
------------
 t
(1 row)

SELECT _timescaledb_internal.enterprise_enabled();
 enterprise_enabled 
--------------------
 t
(1 row)

\set ON_ERROR_STOP 1
\c :TEST_DBNAME :ROLE_SUPERUSER
SELECT _timescaledb_internal.current_license_key();
                                                                            current_license_key                                                                             
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 E1eyJlbmRfdGltZSI6IjIwMTgtMTAtMDEgKzAwMDAiLCAic3RhcnRfdGltZSI6IjIwMTgtMDktMDEgKzAwMDAiLCAiaWQiOiI0OTBGQjI2MC1BMjkyLTRBRDktOUFBMi0wMzYwODM1NzkxQjgiLCAia2luZCI6InRyaWFsIn0K
(1 row)

SELECT _timescaledb_internal.tsl_loaded();
 tsl_loaded 
------------
 t
(1 row)

SELECT _timescaledb_internal.enterprise_enabled();
 enterprise_enabled 
--------------------
 t
(1 row)

\set ON_ERROR_STOP 0
SET timescaledb.license_key='ApacheOnly';
ERROR:  invalid value for parameter "timescaledb.license_key": "ApacheOnly"
SELECT _timescaledb_internal.current_license_key();
                                                                            current_license_key                                                                             
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 E1eyJlbmRfdGltZSI6IjIwMTgtMTAtMDEgKzAwMDAiLCAic3RhcnRfdGltZSI6IjIwMTgtMDktMDEgKzAwMDAiLCAiaWQiOiI0OTBGQjI2MC1BMjkyLTRBRDktOUFBMi0wMzYwODM1NzkxQjgiLCAia2luZCI6InRyaWFsIn0K
(1 row)

SELECT _timescaledb_internal.tsl_loaded();
 tsl_loaded 
------------
 t
(1 row)

SELECT _timescaledb_internal.enterprise_enabled();
 enterprise_enabled 
--------------------
 t
(1 row)

\set ON_ERROR_STOP 1
SET timescaledb.license_key='CommunityLicense';
SELECT _timescaledb_internal.current_license_key();
 current_license_key 
---------------------
 CommunityLicense
(1 row)

SELECT _timescaledb_internal.tsl_loaded();
 tsl_loaded 
------------
 t
(1 row)

SELECT _timescaledb_internal.enterprise_enabled();
 enterprise_enabled 
--------------------
 f
(1 row)

select * from timescaledb_information.license;
  edition  | expired | expiration_time 
-----------+---------+-----------------
 community | f       | infinity
(1 row)

SET timescaledb.license_key=Default;
SELECT _timescaledb_internal.current_license_key();
                                                                            current_license_key                                                                             
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 E1eyJlbmRfdGltZSI6IjIwMTgtMTAtMDEgKzAwMDAiLCAic3RhcnRfdGltZSI6IjIwMTgtMDktMDEgKzAwMDAiLCAiaWQiOiI0OTBGQjI2MC1BMjkyLTRBRDktOUFBMi0wMzYwODM1NzkxQjgiLCAia2luZCI6InRyaWFsIn0K
(1 row)

SELECT _timescaledb_internal.tsl_loaded();
 tsl_loaded 
------------
 t
(1 row)

SELECT _timescaledb_internal.enterprise_enabled();
 enterprise_enabled 
--------------------
 t
(1 row)

select * from timescaledb_information.license;
  edition   | expired |       expiration_time        
------------+---------+------------------------------
 enterprise | t       | Sun Sep 30 17:00:00 2018 PDT
(1 row)

